DB.java

```diff
@@ -294,6 +294,85 @@ else if(i==0)
 		
 		return result;
 	}
+	
+	public boolean check_comments_popular(String ip, int id, int a) throws SQLException
+	{
+		connect();
+		
+		boolean result = false;
+		int i = 0;
 
-}
+		
+
+		String sql = "insert into comments_ip(ip, id, prefer) values(?, ?, ?)";
+		String sql2 = "select * from comments_ip where ip=? and id=?";
+		String sql3 = "update comments_ip set prefer=? where ip=? and id=?";
+		
+		PreparedStatement ps2 = con.prepareStatement(sql2);

```

Comment:
@mjhmohamed 

You referred to `con`.  Where is `con `defined?

Hui

Reply:
`con` is a global object for the sql connection defined as an instance of the class, so it can be accessed from different methods in the same class.

IBRAHIM.